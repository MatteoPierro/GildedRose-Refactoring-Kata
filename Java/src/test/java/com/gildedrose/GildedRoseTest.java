package com.gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class GildedRoseTest {

    @Test
    void goldenMaster() {
        String[] names = {
                "foo",
                "Aged Brie",
                "Backstage passes to a TAFKAL80ETC concert",
                "Sulfuras, Hand of Ragnaros"
        };
        Integer[] sellIns = {-1, 0, 10, 11, 12};
        Integer[] qualities = {-1, 0, 1, 49, 50, 51};
        CombinationApprovals.verifyAllCombinations(
                this::updateQuality,
                names,
                sellIns,
                qualities
        );
    }

    @Test
    void conjuredItem() {
        Approvals.verify(updateQuality("Conjured Pot", 2, 10));
    }

    private Item updateQuality(String name, int sellIn, int quality) {
        Item item = new Item(name, sellIn, quality);
        Item[] items = new Item[]{item};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return item;
    }

}
