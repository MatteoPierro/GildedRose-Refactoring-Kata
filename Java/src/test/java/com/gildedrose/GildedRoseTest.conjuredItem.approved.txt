Conjured Pot, 1, 8